frontend/public/components/catalog.jsx

```diff
@@ -0,0 +1,209 @@
+      filterValue: '',
+      items: [],
+      filteredItems: [],
+    };
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const {clusterserviceclasses, namespace } = this.props;
+    const {filterValue} = this.state;
+    if (namespace !== prevProps.namespace || clusterserviceclasses !== prevProps.clusterserviceclasses) {
+      this.createCatalogData();
+    } else if (filterValue !== prevState.filterValue) {
+      const filteredItems = this.filterItems(this.state.items);
+      this.setState({
+        filteredItems
+      });
+    }
+  }
+
+  filterItems(items) {
+    const {filterValue} = this.state;
+
+    if (!filterValue) {
+      return items;
+    }
+
+    const filterString = filterValue.toLowerCase();
+    return _.filter(items, item => {
+      return fuzzy(filterString, _.get(item, 'metadata.name', ''));
+    });
+  }
+
+  createCatalogData() {
+    const {clusterserviceclasses, imagestreams, loaded} = this.props;
+
+    if (!loaded) {
+      return;
+    }
+
+    const clusterServiceClassItems = this.normalizeClusterServiceClasses(clusterserviceclasses.data, 'ClusterServiceClass');
+    const imagestreamsItems = this.normalizeImagestreams(imagestreams.data, 'ImageStream');
+
+    const items = [
+      ...clusterServiceClassItems,
+      ...imagestreamsItems
+    ];
+
+    const filteredItems = this.filterItems(items);
+    this.setState({items, filteredItems});
+  }
+
+  normalizeClusterServiceClasses(serviceClasses, kind) {
+    return _.map(serviceClasses, serviceClass => {
+      const tileName = serviceClassDisplayName(serviceClass);
+      const tileIcon = <ClusterServiceClassIcon serviceClass={serviceClass} />;
+      const tileDescription = _.get(serviceClass, 'spec.description');
+      const tileVendor = _.get(serviceClass, 'spec.externalMetadata.providerDisplayName');
+
+      return {
+        resource: serviceClass,
+        kind,
+        tileName,
+        tileIcon,
+        tileDescription,
+        tileVendor
+      };
+    });
+  }
+
+  normalizeImagestreams(imagestreams, kind) {
+    imagestreams = _.map(imagestreams, imagestream => {
+      // A valid specTag has a 'builder' tag. no 'hidden' tag, and exists in 'status.tags'.
+      const builderSpecTagName = getBuilderSpecTagName(imagestream);
+      if (builderSpecTagName) {
+        const tileName = _.get(imagestream, ['metadata', 'annotations', 'openshift.io/provider-display-name']) || imagestream.metadata.name;
+        // TODO: icon, description and vendor should be based on selected version?
+        const tileIcon = <span
+          className={`icon ${normalizeIconClass(getImageStreamTagAnnotation(imagestream, 'iconClass', builderSpecTagName))}`}
+          aria-hidden="true">
+        </span>;
+        const tileDescription = getImageStreamTagAnnotation(imagestream, 'description', builderSpecTagName);
+        const tileVendor = getImageStreamTagAnnotation(imagestream, 'openshift.io/provider-display-name', builderSpecTagName);
```

Comment:
Let's call this provider instead of vendor. (I know it's not what we did in the previous catalog.)